app/graphql/schema/mutations/delete-file-metadata-by-blob-reference.js

```diff
@@ -0,0 +1,50 @@
+const cosmos = require('../../../cosmos')
+const { cosmosConfig } = require('../../../config')
+
+const deleteFileMetadataByBlobReference = async (_root, args, context) => {
+  const { filesDatabase } = await cosmos()
+
+  const querySpec = {
+    query: 'SELECT * FROM c WHERE c.metadata.blobReference = @blobReference',
+    parameters: [
+      { name: '@blobReference', value: args.blobReference }
+    ]
+  }
+
+  const { resources } = await filesDatabase
+    .container(cosmosConfig.filesContainer)
+    .items.query(querySpec)
+    .fetchAll()
+
+  if (resources.length === 0) {
+    return {
+      status: {
+        code: 404,
+        success: false,
+        message: 'Metadata not found'
+      }
+    }
+  }
+
+  const fileMetadata = resources[0]
+
+  const response = await filesDatabase
+    .container(cosmosConfig.filesContainer)
+    .item(fileMetadata.id, fileMetadata.partitionKey)
```

Comment:
Might seem confusing as to why I've put `fileMetadata.id` and `fileMetadata.partitionKey` instead of `fileMetadata.blobReference` since this mutation deletes metadata by providing an existing `blobReference` as an argument. The uniqueness of all items in a Cosmos database/container is based on its `id` and `partitionKey` and therefore these are needed to accurately delete an item from the database without any risk of ambiguity. I found that when I used `fileMetadata.blobReference` a `404` error was thrown suggesting the item could not be found/didn't exist in the database.

The mutation works first by taking the `blobReference` entered as the input, then querying the database to find all the items which might match that `blobReference`. Then it runs the `delete()` method provided by the Cosmos SDK, but it will specifically do it on the `id` and `partitionKey` that correspond to the provided `blobReference`.

Reply:
Makes perfect sense 😄